src/test/java/core/basesyntax/model/PurchaseOperationTest.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.model;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Warehouse;
+import core.basesyntax.model.entities.Fruit;
+import core.basesyntax.model.impl.PurchaseOperation;
+import org.junit.Test;
+
+public class PurchaseOperationTest {
+    private static final Warehouse<Fruit> WAREHOUSE = new Warehouse<>();
+
+    @Test
+    public void executeBalanceOperationTest() {
+        PurchaseOperation<Fruit> purchaseOperation = new PurchaseOperation<>(WAREHOUSE);
+        purchaseOperation.execute(new Fruit("test"), 10);
+        assertEquals(1, WAREHOUSE.getStorage().size());
```

Comment:
will we have 10 fruit if the warehouse is empty?

Reply:
Could you please clarify?

Reply:
in this test you have an empty storage
then you attempt to purchase 10 fruit - your code should throw an exception in this case